### PR TITLE
Add 1.14 branches for the publishing-bot

### PR DESCRIPTION
**Pull request by dims**
Change-Id: Id34d1f12361e587e02f7dba077a9600618737878

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Line up the upcoming release

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -194,6 +222,18 @@ rules:
...
+      dir: staging/src/k8s.io/apiserver
+    name: release-1.14
+    go: 1.11.2
+    dependencies:
+      - repository: apimachinery
+        branch: release-1.14
+      - repository: api
+        branch: release-1.14
+      - repository: client-go
+        branch: release-10.0
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: this needs to be release-11.0

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -124,6 +142,16 @@ rules:
       branch: release-1.13
     - repository: api
       branch: release-1.13
+  - source:
+      branch: release-1.14
+      dir: staging/src/k8s.io/client-go
+    name: release-10.0
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: release-11.0

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (nikhita) [reply staging/publishing/rules.yaml]: (ditto everywhere)

**Comment by nikhita** [comment]
@dims thanks for this! This skipped my mind -- we should probably write down the list of things needed to be done for each release somewhere. I have most of it scattered across issues so I'll try to do that later.

File: staging/publishing/rules.yaml (hunk 1 above)
Comment (dims) [reply staging/publishing/rules.yaml]: Done!

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (dims) [reply staging/publishing/rules.yaml]: Done!

File: staging/publishing/rules.yaml [hunk 3]
```diff
@@ -26,6 +26,11 @@ rules:
       dir: staging/src/k8s.io/code-generator
     name: release-1.13
     go: 1.11.2
+  - source:
+      branch: release-1.14
+      dir: staging/src/k8s.io/code-generator
+    name: release-1.14
+    go: 1.11.2
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: looks like this should be 1.11.5

https://github.com/kubernetes/kubernetes/commit/e273db831c3bdd74d9bb4590e6d808ff75578467

File: staging/publishing/rules.yaml (hunk 2 above)
Comment (nikhita) [reply staging/publishing/rules.yaml]: oops, looks like this got left out :)

File: staging/publishing/rules.yaml (hunk 3 above)
Comment (dims) [reply staging/publishing/rules.yaml]: ah. ack

File: staging/publishing/rules.yaml [hunk 4]
```diff
@@ -845,6 +1051,18 @@ rules:
...
+    name: release-1.14
+    go: 1.11.5
+    dependencies:
+      - repository: api
+        branch: release-1.14
+      - repository: apimachinery
+        branch: release-1.14
+      - repository: client-go
+        branch: release-10.0
 - destination: node-api
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: node-api + repos below this also need the release-1.14 branch

**Comment by dims** [comment]
Thanks for the super quick reviews @nikhita hopefully i fixed all the issues so far!

File: staging/publishing/rules.yaml [hunk 5]
```diff
@@ -845,31 +1051,69 @@ rules:
...
+      - repository: client-go
+        branch: release-11.0
 - destination: node-api
   library: true
   branches:
   - source:
       branch: master
       dir: staging/src/k8s.io/node-api
     name: master
+    go: 1.11.5
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: We have the default go version defined here - https://github.com/kubernetes/publishing-bot/blob/master/pkg/golang/install.go#L32 so I don't think we need this.

**Comment by nikhita** [comment]
@dims last comment but lgtm otherwise. 

yaml is hard. :)

**Comment by sttts** [comment]
component-base is missing, isn't it?

@dims can you do a smoke test run?

**Comment by dims** [comment]
@sttts how do you do that usually? kubernetes-nightly or kubernetes configuration? What configuration edits are needed?

**Comment by sttts** [comment]
@dims second copy of the main config, but with dry-run. Nightly would work as well. I hope we can pass a rules URL of the branch.

**Comment by dims** [comment]
Here's the diff i am using:
```
diff --git a/configs/kubernetes-nightly b/configs/kubernetes-nightly
index 754c43ad959a..d0f3400e4b3e 100644
--- a/configs/kubernetes-nightly
+++ b/configs/kubernetes-nightly
@@ -1,4 +1,4 @@
-DOCKER_REPO = sttts/k8s-publishing-bot-nightly
+DOCKER_REPO = dims/k8s-publishing-bot-nightly
 NAMESPACE = k8s-publishing-bot-nightly
 SCHEDULE = * */4 * * *
 INTERVAL = 14400
diff --git a/configs/kubernetes-nightly-configmap.yaml b/configs/kubernetes-nightly-configmap.yaml
index f2cb0e54c9ef..c26fca64ed0f 100644
--- a/configs/kubernetes-nightly-configmap.yaml
+++ b/configs/kubernetes-nightly-configmap.yaml
@@ -7,6 +7,6 @@ data:
     source-org: kubernetes
     source-repo: kubernetes
     target-org: kubernetes-nightly
-    rules-file: https://raw.githubusercontent.com/kubernetes/kubernetes/master/staging/publishing/rules.yaml
+    rules-file: https://raw.githubusercontent.com/dims/kubernetes/prepare-publishing-bot-for-1.14-branch-%2B244-%E2%88%920/staging/publishing/rules.yaml
     # github-issue: 56876
-    dry-run: false
+    dry-run: true
```

Here's what i am running:
```
hack/fetch-all-latest-and-push.sh kubernetes-nightly
make build-image push-image CONFIG=configs/kubernetes-nightly
make run CONFIG=configs/kubernetes-nightly
```

watching the logs now...

**Comment by dims** [comment]
@sttts @nikhita ended in failure - http://paste.openstack.org/raw/746610/

File: staging/publishing/rules.yaml [hunk 6]
```diff
@@ -26,6 +26,11 @@ rules:
       dir: staging/src/k8s.io/code-generator
     name: release-1.13
     go: 1.11.2
+  - source:
+      branch: release-1.14
+      dir: staging/src/k8s.io/code-generator
+    name: release-1.14
+    go: 1.11.5
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: we use go 1.12 now https://github.com/kubernetes/kubernetes/pull/74632

sorry for so much back-and-forth, @dims :see_no_evil:

File: staging/publishing/rules.yaml [hunk 7]
```diff
@@ -403,6 +479,22 @@ rules:
       branch: release-1.13
     required-packages:
     - k8s.io/code-generator
+  - source:
+      branch: release-1.14
+      dir: staging/src/k8s.io/sample-controller
+    name: release-1.13
```
Comment (nikhita) [inline staging/publishing/rules.yaml]: should be release-1.14

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

/hold
will let @sttts cancel the hold

**Pull request merged**